local/nginx/include/routes.conf

```diff
@@ -1,29 +1,145 @@
 location ~ ^/content/(.*)$ {
+
+    set_by_lua_file $secret "/etc/nginx/lua/read_public_key.lua";
+    set_by_lua_file $missing_token "/etc/nginx/lua/missing_token.lua";
+    set_by_lua_file $invalid_token "/etc/nginx/lua/invalid_token.lua";
+    set_by_lua_file $jwt_req_zone "/etc/nginx/lua/parse_jwt_as_token.lua" "content/" $1;
+
+    proxy_pass_request_headers  on;
+    proxy_connect_timeout       600;
+    proxy_send_timeout          600;
+    proxy_read_timeout          600;
+    send_timeout                600;
+    proxy_set_header X-Forwarded-Host  $host;
+
+    # Hacky workaround as nginx does not allow redirects to named locations under if conditional 
+    error_page 400 = @uniqueLimitContentServer;
+
+    if ($missing_token) {
+        # It does not return an error code
+        return 400;
+        break;
+    }
+
+    if ($invalid_token) {
+        return 401 "Access denied because JWT is expired or invalid";
+        break;
+    }
```

Comment:
I'm thinking if the default behavior should be to bounce the requests and deny the access or simply putting the request in a "shared bucket" to prevent unauthorized DDoS. From what I am reading this implements the first, deny access if missing JWT or invalid. Is it like that?

Reply:
We've discussed this with @jmoguilevsky. The first implementation was:
- If there is no token in the Cookie header => send it to the shared bucket.
- If the token is invalid => send it to the shared bucket.
- If the token is valid => send it to its custom bucket.

Anyway, then we've realized that as a client you don't have an easy way to identify when a token has been invalidated if you are always sent to the shared bucket. That's why we thought that the solution for it was to return an error when the token was present but invalid, this way the client can implement the logic to retry generating a new JWT.

Reply:
After discussing this with @menduz & @jmoguilevsky  we've decided that when the JWT is invalid then we are sending the traffic through the same rate limit as the not identified requests, but we are removing the cookie of the JWT in the header "Set-Cookie"